view/base/web/js/model/address_validation_core.js

```diff
@@ -93,8 +89,46 @@ function (ko, $) {
             }
         },
 
+        isSuggestedAddress: function (address) {
+            var self = this;
+            var isSuggested = false;
+            var suggestedAddresses = this.suggestedAddresses();
+            suggestedAddresses.forEach(function(suggestedAddress, index) {
+                if (self.addressMatches(self.formatAddress(address), self.formatSuggestedAddress(suggestedAddress.address))) {
+                    isSuggested = true;
+                }
+            });
+            return isSuggested;
+        },
+
         updateSuggestedAddresses: function (addr) {
             this.suggestedAddresses(addr);
+        },
+
+        formatAddress: function (address) {
+            var formattedAddress = {
+                'street0': address.street[0],
+                'city': address.city,
+                'region': address.region_id,
+                'country': address.country_id,
+                'postcode': address.postcode
+            };
+            return formattedAddress;
+        },
+
+        formatSuggestedAddress: function (suggestedAddress) {
+            var formattedAddress = {
+                'street0': suggestedAddress.street[0],
+                'city': suggestedAddress.city,
+                'region': suggestedAddress.regionId,
+                'country': suggestedAddress.countryId,
+                'postcode': suggestedAddress.postcode
+            };
+            return formattedAddress;
+        },
+
+        addressMatches: function (address, addressToCompare) {
+            return JSON.stringify(address) === JSON.stringify(addressToCompare);
```

Comment:
In this particular use case, because they same functions are used to format the address when they are stored, they will always have the same order. I think building a function to do a deep comparison isn't necessary here.